File: src/main/java/uk/laxd/dndSimulator/action/DamageModifier.java
package uk.laxd.dndSimulator.action;

import uk.laxd.dndSimulator.dice.Die;

import java.util.Arrays;

public class DamageModifier {

    private int damageBonus;
    private Die[] diceBonus;

    public DamageModifier(int damageBonus, Die... diceBonus) {
        this.damageBonus = damageBonus;
        this.diceBonus = diceBonus;
    }

    public int getDamageBonus() {
        return damageBonus;
    }

    public Die[] getDiceBonus() {
        return diceBonus;
    }

    public int getTotalDamageBonus() {
        return damageBonus + Arrays.stream(diceBonus).mapToInt(Die::roll).sum();
    }
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/action/DamageModifier.kt
package uk.laxd.dndSimulator.action

import uk.laxd.dndSimulator.dice.Die

class DamageModifier(private val damageBonus: Int, private val diceBonus: Collection<Die>) {

    val totalDamageBonus: Int
        get() = damageBonus + diceBonus.stream().mapToInt { obj: Die -> obj.roll() }.sum()

}